tests/gumball-machine-test.ts

```diff
@@ -511,42 +547,208 @@ describe("gumball-machine", () => {
+      });
+      it("Can destroy gumball machine and reclaim lamports", async () => {
+        await destroyGumballMachine(gumballMachineAcctKeypair, creatorAddress);
+      });
     });
 
-    it("Can update config lines", async () => {
-      await updateConfigLines(payer, gumballMachineAcctKeypair.publicKey, GUMBALL_MACHINE_ACCT_SIZE, GUMBALL_MACHINE_ACCT_CONFIG_INDEX_ARRAY_SIZE, GUMBALL_MACHINE_ACCT_CONFIG_LINES_SIZE, Buffer.from("aaavnpwncgchwnbqfpbtdlcpdaaa"), new BN(0));
-    });
-    it("Can update gumball header", async () => {
-      const newGumballMachineHeader: InitGumballMachineProps = {
-        urlBase: Buffer.from("https://arweave.net/bzdjillretjcraaxawlnhqrhmexzbsixyajrlzhfcvcc"),
-        nameBase: Buffer.from("wmqeslreeondhmcmtfebrwqnqcoasbye"),
-        symbol: Buffer.from("wmqeslreeondhmcmtfebrwqnqcoasbye"), 
-        sellerFeeBasisPoints: 50,
-        isMutable: false,
-        retainAuthority: false,
-        price: new BN(100),
-        goLiveDate: new BN(5678.0),
-        mint: wrappedSolPubKey,          // Cannot be modified after init
-        botWallet: Keypair.generate().publicKey,
-        authority: Keypair.generate().publicKey,
-        collectionKey: payer.publicKey,  // Cannot be modified after init
-        creatorAddress: payer.publicKey, // Cannot be modified after init
-        extensionLen: new BN(28),        // Cannot be modified after init
-        maxMintSize: new BN(15),
-        maxItems: new BN(250)            // Cannot be modified after init
-      };
-      await updateHeaderMetadata(payer, gumballMachineAcctKeypair.publicKey, GUMBALL_MACHINE_ACCT_SIZE, newGumballMachineHeader);
-    })
-    it("Can destroy gumball machine and reclaim lamports", async () => {
-      await destroyGumballMachine(gumballMachineAcctKeypair, payer);
-    })
-    it("Can dispense single NFT", async () => {
-      await dispenseCompressedNFT(new BN(1), payer, gumballMachineAcctKeypair, merkleRollKeypair, noncePDAKey);
+    describe("buyer instructions", async () => {
+      let nftBuyer: Keypair;
+      beforeEach(async () => {
+        creatorAddress = Keypair.generate();
+        // Give creator enough funds to produce accounts for gumball-machine
+        await GumballMachine.provider.connection.confirmTransaction(
+          await GumballMachine.provider.connection.requestAirdrop(creatorAddress.publicKey, 4 * LAMPORTS_PER_SOL),
+          "confirmed"
+        );
+        gumballMachineAcctKeypair = Keypair.generate();
+        merkleRollKeypair = Keypair.generate();
+        nftBuyer = Keypair.generate();
+      });
+      it("Can dispense single NFT paid in sol", async () => {
+        let creatorPaymentWallet = Keypair.generate();
+
+        baseGumballMachineHeader = {
+          urlBase: Buffer.from("https://arweave.net/Rmg4pcIv-0FQ7M7X838p2r592Q4NU63Fj7o7XsvBHEEl"),
+          nameBase: Buffer.from("zfgfsxrwieciemyavrpkuqehkmhqmnim"),
+          symbol: Buffer.from("pehjjqmrjpfcnttlierdqkxjueqjqjsf"), 
```

Comment:
LOL how many characters are these?
Pretty sure that symbol < 16 chars for a normal NFT, so decompressing this probably won't work 😆

Reply:
I think Symbol on the TokenMetadata struct actually has a 32 byte max

Reply:
JK I'm wrong it's a Vec<u8> with max len of 8.

Reply:
yuck 🤮

Reply:
yeah these are just random 32 char strings. If we want to ensure decompress-ability, let's change this https://github.com/jarry-xiao/candyland/blob/566cd5ae3ce596f29585c5317f0a55cd976b0f40/programs/gumball-machine/src/state/mod.rs#L18 to have max size of 8?